apps/system/test/marionette/notification_events_test.js

```diff
@@ -369,51 +378,28 @@ marionette('Notification events', function() {
     }, [details]);
 
     client.switchToFrame();
-    client.apps.close(EMAIL_APP);
-
-    // after closing live handlers should be lost, the callbacks too
-    client.apps.launch(EMAIL_APP);
-    client.apps.switchToApp(EMAIL_APP);
-
-    client.executeScript(fs.readFileSync(
-      __dirname + '/lib/fake_moz_set_message_handler.js', 'utf8'));
-
-    // fake mozSetMessageHandler
-    client.executeScript(function() {
-      window.wrappedJSObject.mozSetMessageHandler('notification');
-    });
+    client.apps.close(MESSAGE_HANDLER_APP);
+    client.apps.launch(MESSAGE_HANDLER_APP);
 
     client.switchToFrame();
 
-    client.executeScript(function(manifest) {
-      // get notifications
-      var container =
-        document.getElementById('desktop-notifications-container');
-      var selector = '[data-manifest-u-r-l="' + manifest + '"]';
-      var node = container.querySelectorAll(selector)[0];
-
-      // simulate tapping
-      var event = document.createEvent('CustomEvent');
-      event.initCustomEvent('mozContentNotificationEvent', true, true, {
-        type: 'desktop-notification-click',
-        id: node.dataset.notificationId
-      });
-      window.dispatchEvent(event);
-    }, [EMAIL_APP_MANIFEST]);
+    notificationList.waitForNotificationCount(1);
+    notificationList.refresh();
+    // clicks away all monitored notifications
+    // -> mozSetMessageHandler should get the corresponding click events
+    notificationList.clear();
```

Comment:
Actually, this clicks the "Clear All" button, which swipes away all the notifications currently in the system tray. So the system message you receive in your app is the "close" event rather than click. I would prefer just manually tapping the notification container, but this works too so long as you update the comments to say what's happening.